### PR TITLE
Right Truncation in NSSP Demo

### PR DESCRIPTION
**Pull request by damonbayer**
This PR enables model fitting with right-truncation estimates provided by NNH.

### CODE REVIEW

File: pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            )
+        else:
+            right_truncation_multiplier = 1
+
+        latent_hospital_admissions = (
+            potential_latent_hospital_admissions
+            * ihr
+            * hosp_wday_effect
+            * self.state_pop
+            * right_truncation_multiplier
```
Comment (damonbayer) [inline pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py:213]: This is new.

File: pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                n_datapoints - right_truncation_multiplier_tail.shape[0]
+            )
+            right_truncation_multiplier = jnp.pad(
+                right_truncation_multiplier_tail,
+                (n_points_to_prepend, 0),
+                mode="constant",
+                constant_values=(1, 0),
+            )
+        else:
+            right_truncation_multiplier = 1
```
Comment (damonbayer) [inline pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py:206]: This is new.

Comment (damonbayer) [inline pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py:206]: This is new.

Comment (dylanhmorris) [inline pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py:206]: 1. Name the variable `fraction_already_reported` or similar rather than `right_truncation_multiplier`?
2. Might it be more readable to construct it by creating a `truncated` mask and then doing
```python
fraction_already_reported = jnp.where(
    truncated,
    right_truncation_multiplier_tail
    1)
```
?

File: pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py (hunk 1 above)
Comment (dylanhmorris) [inline pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py:213]: Would be nice to have the separate `predicted_final_reported_admissions` and `predicted_current_reported_admissions` where

```python
 predicted_final_reported_admissions = (
   potential_latent_hospital_admissions
   * ihr
   * hosp_wday_effect
   * self.state_pop)
predicted_current_reported_admissions = (predicted_final_reported_admissions 
   * right_truncation_multiplier)
```
That way we get a nowcast of the final reports for free.

**Review by dylanhmorris — CHANGES_REQUESTED** [review]
Looks very good. Thanks @damonbayer. A few minor things.

**Comment by damonbayer** [comment]
Thanks @dylanhmorris. Will address your comments and probably merge this back into the original model with optional right truncation.

File: pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py (hunk 2 above)
Comment (damonbayer) [reply pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py:206]: I don't think this works because `truncated` is longer than `right_truncation_multiplier_tail`?

File: pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        i0 = self.infection_initialization_process()
+
+        eta_sd = self.eta_sd_rv()
+        autoreg_rt = self.autoreg_rt_rv()
+        log_r_mu_intercept = self.log_r_mu_intercept_rv()
+        rt_init_rate_of_change_rv = DistributionalVariable(
+            "rt_init_rate_of_change",
+            dist.Normal(0, eta_sd / jnp.sqrt(1 - jnp.pow(autoreg_rt, 2))),
+        )
+        rt_init_rate_of_change = rt_init_rate_of_change_rv()
```
[the remark this answers is not in the document]
Comment (damonbayer) [reply pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py]: addressed in https://github.com/CDCgov/pyrenew-covid-wastewater/pull/17/commits/3328e14fe2bc9cd66013688a740d73d864bb3cb9

File: pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py (hunk 2 above)
Comment (dylanhmorris) [reply pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py]: You're totally right, apologies. Do think the variable name suggestion makes sense.

Comment (damonbayer) [reply pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py]: Changed the name in https://github.com/CDCgov/pyrenew-covid-wastewater/pull/17/commits/87186c088d6a20b2648587d790173db3424d2afc
